src/raftstore/store/peer.rs

```diff
@@ -889,6 +900,15 @@ impl Peer {
         }
 
         let apply_snap_result = self.mut_store().post_ready(invoke_ctx);
+        if apply_snap_result.is_some() && self.peer.get_is_learner() {
+            // The peer may change from learner to voter after snapshot applied.
+            self.peer = self.region()
```

Comment:
seem you can't add a test for it - changing the learner flag in the snapshot not in AddNode.

Reply:
`AddNode` with an exists learner is just a promotion. After we promote the learner and restart it, leader will send a snapshot to make it as a voter.

Reply:
Maybe we need to log it.